clippy_lints/src/methods/chunks_exact_with_const_size.rs

```diff
@@ -0,0 +1,91 @@
+use clippy_utils::consts::{ConstEvalCtxt, Constant};
+use clippy_utils::diagnostics::span_lint_and_then;
+use clippy_utils::msrvs::{self, Msrv};
+use clippy_utils::source::snippet_with_applicability;
+use clippy_utils::sym;
+use rustc_errors::Applicability;
+use rustc_hir::{Expr, Node, PatKind};
+use rustc_lint::LateContext;
+use rustc_span::{Span, Symbol};
+
+use super::CHUNKS_EXACT_WITH_CONST_SIZE;
+
+pub(super) fn check(
+    cx: &LateContext<'_>,
+    recv: &Expr<'_>,
+    arg: &Expr<'_>,
+    expr: &Expr<'_>,
+    call_span: Span,
+    method_name: Symbol,
+    msrv: Msrv,
+) {
+    // Check if receiver is slice-like
+    if !cx.typeck_results().expr_ty_adjusted(recv).peel_refs().is_slice() {
+        return;
+    }
```

Comment:
This doesn't actually constrain the method correctly. You want to for a single reference to a slice.